x/merkledb/db.go

```diff
@@ -954,22 +970,22 @@ func (db *merkleDB) VerifyChangeProof(
 	// Find the greatest key in [proof.KeyChanges]
 	// Note that [proof.EndProof] is a proof for this key.
 	// [largestPath] is also used when we add children of proof nodes to [trie] below.
-	largestPath := maybe.Bind(end, newPath)
+	largestPath := maybe.Bind(end, db.newPath)
 	if len(proof.KeyChanges) > 0 {
 		// If [proof] has key-value pairs, we should insert children
 		// greater than [end] to ancestors of the node containing [end]
 		// so that we get the expected root ID.
-		largestPath = maybe.Some(newPath(proof.KeyChanges[len(proof.KeyChanges)-1].Key))
+		largestPath = maybe.Some(db.newPath(proof.KeyChanges[len(proof.KeyChanges)-1].Key))
 	}
 
 	// Make sure the end proof, if given, is well-formed.
-	if err := verifyProofPath(proof.EndProof, largestPath.Value()); err != nil {
+	if err := verifyProofPath(proof.EndProof, largestPath); err != nil {
 		return err
 	}
 
-	keyValues := make(map[path]maybe.Maybe[[]byte], len(proof.KeyChanges))
+	keyValues := make(map[Path]maybe.Maybe[[]byte], len(proof.KeyChanges))
```

Comment:
There are a bunch of places we had a `map[path]something` which are now `map[Path]something`. Are we sure this is safe everywhere? `Path` is a struct, not a `string`, so the lookup behavior is different when we're using it as a map key.

Reply:
Structs are treated as a collection of their fields, so as long as the fields can be map keys, it all works.  This is relying upon the fact that the pathconfig pointers always point to the same config for a given branchfactor.